third_party/blink/renderer/core/brave_page_graph/page_graph.cc

```diff
@@ -1457,7 +1514,9 @@ void PageGraph::RegisterRequestStartFromElm(const DOMNodeId node_id,
 
   // We should know about the node thats issuing the request.
   NodeHTMLElement* const requesting_node = GetHTMLElementNode(node_id);
-  DoRegisterRequestStart(request_id, requesting_node, normalized_url,
+  blink::Node* node = blink::DOMNodeIds::NodeForId(node_id);
+  FrameId frame_id = GetFrameId(node);
```

Comment:
I'm getting CHECK failure here on nytimes.com:

```
[56476:87136:0411/175747.600:VERBOSE1:page_graph.cc(1510)] RegisterRequestStartFromElm) node id: 5819, request id: 820, url: "https://www.youtube.com/s/player/7ebf4817/www-embed-player.vflset/www-embed-player.js", type: "Script"
[56476:87136:0411/175747.600:VERBOSE1:page_graph.cc(1159)] GetHTMLElementNode) node id: 5819   
[56476:87136:0411/175747.600:FATAL:page_graph.cc(213)] Check failed: execution_context.
```

```
 	base.dll!logging::LogMessage::HandleFatal(unsigned __int64 stack_start, const std::__Cr::basic_string<char,std::__Cr::char_traits<char>,std::__Cr::allocator<char>> & str_newline) Line 1033	C++
 	[Inline Frame] base.dll!logging::LogMessage::Flush::<lambda_0>::operator()() Line 737	C++
 	[Inline Frame] base.dll!absl::cleanup_internal::Storage<`lambda at ..\..\base\logging.cc:735:40'>::InvokeCallback() Line 87	C++
 	[Inline Frame] base.dll!absl::Cleanup<absl::cleanup_internal::Tag,`lambda at ..\..\base\logging.cc:735:40'>::~Cleanup() Line 106	C++
 	base.dll!logging::LogMessage::Flush() Line 920	C++
 	base.dll!logging::LogMessage::~LogMessage() Line 695	C++
 	[Inline Frame] base.dll!logging::`anonymous namespace'::CheckLogMessage::~CheckLogMessage() Line 185	C++
 	base.dll!logging::`anonymous namespace'::CheckLogMessage::~CheckLogMessage() Line 180	C++
 	[Inline Frame] base.dll!std::__Cr::default_delete<logging::LogMessage>::operator()(logging::LogMessage * __ptr) Line 68	C++
 	[Inline Frame] base.dll!std::__Cr::unique_ptr<logging::LogMessage,std::__Cr::default_delete<logging::LogMessage>>::reset(logging::LogMessage * __p) Line 279	C++
 	base.dll!logging::CheckError::~CheckError() Line 343	C++
>	[Inline Frame] blink_core.dll!blink::`anonymous namespace'::GetFrameId(blink::Node * node) Line 213	C++
 	blink_core.dll!blink::PageGraph::RegisterRequestStartFromElm(const int node_id, const unsigned __int64 request_id, const blink::KURL & url, const WTF::String & resource_type) Line 1518	C++
 	blink_core.dll!blink::PageGraph::WillSendRequest(blink::ExecutionContext * execution_context, blink::DocumentLoader * loader, const blink::KURL & fetch_context_url, const blink::ResourceRequest & request, const blink::ResourceResponse & redirect_response, const blink::ResourceLoaderOptions & options, blink::ResourceType resource_type, blink::RenderBlockingBehavior render_blocking_behavior, base::TimeTicks timestamp) Line 499	C++
 	[Inline Frame] blink_core.dll!blink::probe::WillSendRequestImpl::<lambda_55>::operator()(blink::PageGraph * agent) Line 1499	C++
 	[Inline Frame] blink_core.dll!blink::AgentRegistry<blink::PageGraph>::ForEachAgent(const blink::probe::WillSendRequestImpl::<lambda_55> & callable) Line 75	C++
 	blink_core.dll!blink::probe::WillSendRequestImpl(blink::ExecutionContext * param_execution_context, blink::DocumentLoader * param_document_loader, const blink::KURL & fetch_context_url, const blink::ResourceRequest & param_resource_request, const blink::ResourceResponse & redirect_response, const blink::ResourceLoaderOptions & param_resource_loader_options, blink::ResourceType param_resource_type, blink::RenderBlockingBehavior param_render_blocking_behavior, base::TimeTicks timestamp) Line 1498	C++
 	[Inline Frame] blink_core.dll!blink::probe::WillSendRequest(blink::ExecutionContext * param_execution_context, blink::DocumentLoader * param_document_loader, const blink::KURL & fetch_context_url, const blink::ResourceRequest & param_resource_request, const blink::ResourceResponse & redirect_response, const blink::ResourceLoaderOptions & param_resource_loader_options, blink::ResourceType param_resource_type, blink::RenderBlockingBehavior param_render_blocking_behavior, base::TimeTicks timestamp) Line 393	C++
 	blink_core.dll!blink::ResourceLoadObserverForFrame::WillSendRequest(const blink::ResourceRequest & request, const blink::ResourceResponse & redirect_response, blink::ResourceType resource_type, const blink::ResourceLoaderOptions & options, blink::RenderBlockingBehavior render_blocking_behavior, const blink::Resource * resource) Line 164	C++
 	blink_platform.dll!blink::ResourceFetcher::DidLoadResourceFromMemoryCache(blink::Resource * resource, const blink::ResourceRequest & request, bool is_static_data, blink::RenderBlockingBehavior render_blocking_behavior) Line 765	C++
 	blink_platform.dll!blink::ResourceFetcher::RequestResource(blink::FetchParameters & params, const blink::ResourceFactory & factory, blink::ResourceClient * client) Line 1261	C++
 	blink_core.dll!blink::ScriptResource::Fetch(blink::FetchParameters & params, blink::ResourceFetcher * fetcher, blink::ResourceClient * client, v8::Isolate * isolate, blink::ScriptResource::StreamingAllowed streaming_allowed, blink::v8_compile_hints::V8CrowdsourcedCompileHintsProducer * v8_compile_hints_producer, blink::v8_compile_hints::V8CrowdsourcedCompileHintsConsumer * v8_compile_hints_consumer) Line 108	C++
 	blink_core.dll!blink::ClassicPendingScript::Fetch(const blink::KURL & url, blink::Document & element_document, const blink::ScriptFetchOptions & options, blink::CrossOriginAttributeValue cross_origin, const WTF::TextEncoding & encoding, blink::ScriptElementBase * element, blink::FetchParameters::DeferOption defer, blink::scheduler::TaskAttributionInfo * parent_task) Line 110	C++
 	blink_core.dll!blink::ScriptLoader::PrepareScript(blink::ScriptLoader::ParserBlockingInlineOption parser_blocking_inline_option, const WTF::TextPosition & script_start_position) Line 956	C++
 	blink_core.dll!blink::HTMLParserScriptRunner::ProcessScriptElementInternal(blink::Element * script, const WTF::TextPosition & script_start_position) Line 535	C++
 	blink_core.dll!blink::HTMLParserScriptRunner::ProcessScriptElement(blink::Element * script_element, const WTF::TextPosition & script_start_position) Line 302	C++
 	blink_core.dll!blink::HTMLDocumentParser::RunScriptsForPausedTreeBuilder() Line 682	C++
 	[Inline Frame] blink_core.dll!blink::HTMLDocumentParser::CanTakeNextToken(base::TimeDelta & time_executing_script) Line 193	C++
 	blink_core.dll!blink::HTMLDocumentParser::PumpTokenizer() Line 751	C++
 	blink_core.dll!blink::HTMLDocumentParser::PumpTokenizerIfPossible() Line 649	C++
 	blink_core.dll!blink::HTMLDocumentParser::DeferredPumpTokenizerIfPossible(bool from_finish_append, base::TimeTicks schedule_time) Line 629	C++
```

Reply:
Hmm, im not able to reproduce this with the below?

`npm run crawl -- -b <path> -u https://www.nytimes.com -o ~/Desktop/nytimes.graphml -t 30 --debug verbose`

I've tried a good chunk of additional pages on nytimes.com too.   Can you share what command / page you're looking at? Do you see this issue on other sites?

Reply:
I also cannot reproduce it now, but looking into the stacktrace: this is youtube script, it's pulled from the resource cache.

I did use our internal VPN to open the website.

Reply:
tbh I think you should use `DocumentLoader here to get the frame id.